dbms/src/Storages/KVStore/tests/gtest_region_persister.cpp

```diff
@@ -245,6 +248,220 @@ try
 }
 CATCH
 
+
+TEST_F(RegionSeriTest, FlexibleRestore)
+try
+{
+    auto ext_cnt_1 = 0;
+    auto ext_cnt_2 = 0;
+    auto ext_cnt_3 = 1;
+    auto ext_cnt_4 = 2;
+    {
+        auto counter = std::make_shared<int>(0);
+        // V3 store, V3 load, no unrecognized fields
+        auto region = std::make_shared<Region>(createRegionMeta(1001, 1));
+        region->updateRaftLogEagerIndex(5678);
+        FailPointHelper::enableFailPoint(FailPoints::force_region_persist_version, std::make_pair(3, ext_cnt_3));
+        FailPointHelper::enableFailPoint(FailPoints::force_region_persist_extension_field, static_cast<int>(1));
+        const auto path = dir_path + "/region.test";
+        WriteBufferFromFile write_buf(path, DBMS_DEFAULT_BUFFER_SIZE, O_WRONLY | O_CREAT);
+        size_t region_ser_size = std::get<0>(region->serialize(write_buf));
+        write_buf.next();
+        write_buf.sync();
+        ASSERT_EQ(region_ser_size, (size_t)Poco::File(path).getSize());
+
+        FailPointHelper::enableFailPoint(FailPoints::force_region_read_version, static_cast<UInt64>(3));
+        FailPointHelper::enableFailPoint(FailPoints::force_region_read_extension_field, std::make_pair(1, counter));
+        ReadBufferFromFile read_buf(path, DBMS_DEFAULT_BUFFER_SIZE, O_RDONLY);
+        auto new_region = Region::deserialize(read_buf);
+        ASSERT_EQ(new_region->getRaftLogEagerGCRange().first, 5678);
+        ASSERT_REGION_EQ(*new_region, *region);
+        ASSERT_EQ(*counter, 1);
+
+        FailPointHelper::disableFailPoint(FailPoints::force_region_persist_version);
+        FailPointHelper::disableFailPoint(FailPoints::force_region_read_version);
+        FailPointHelper::disableFailPoint(FailPoints::force_region_persist_extension_field);
+        FailPointHelper::disableFailPoint(FailPoints::force_region_read_extension_field);
+    }
+    {
+        auto counter = std::make_shared<int>(0);
+        // Downgrade. V4(whatever) store, V3 load, UNUSED_EXTENSION_NUMBER_FOR_TEST unrecognized.
```

Comment:
```suggestion
        // Downgrade. V4(whatever) store, V2 load, UNUSED_EXTENSION_NUMBER_FOR_TEST unrecognized.
```

Reply:
These tests assume we have Test in V3 and UNUSED_EXTENSION_NUMBER_FOR_TEST in V4